pkg/command/util/fake.go

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2022 Singularity Data
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package util
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	"github.com/singularity-data/risingwave-operator/pkg/testutils"
+
+	"github.com/singularity-data/risingwave-operator/apis/risingwave/v1alpha1"
+)
+
+func NewFakeClient() client.Client {
+	risingwave := &v1alpha1.RisingWave{}
+	c := fake.NewClientBuilder().
+		WithScheme(testutils.Schema).
+		WithObjects(risingwave).
+		Build()
+	return c
+}
+
+var FakeRW = &v1alpha1.RisingWave{
```

Comment:
We'd better hide this object and always use `DeepCopy` to get one. This avoids flaky tests which are caused by accidental modifications.

```go
var fakeRW = xx

func FakeRW() {
   return fakeRW.DeepCopy()
}
```

Reply:
Good idea. I call the `DeepCopy` in the function.  Call `DeepCopy` when using it is better.